### PR TITLE
General Fixes

### PR DESCRIPTION
**Pull request by EthanHindmarsh**
# Fixes / Changes:

#### default_config.ini:
- Change default release channel to STABLE instead of ALL (most users probably shouldn't be on development versions)
- Fix spacing error in colors_enabled comment

#### install.sh:
- remove usage of `function` for consistency
- change confirm to accept y/yes & n/no
- change `read` in confirm to not mangle backslashes

#### README.md:
- change some HTML styles
- Add alt texts to all images for screen readers
- Update `How to Download`
- Create a header under Linux/MacOS instructions called `Automatic Installation` to make it more clear.

#### auth.py:
- Remove googleapiclient.errors HttpError, unused imports
- remove redundant use of 'global' on CURRENTUSER and YOUTUBE

#### community_downloader.py:
- Remove argparse, io, and os, unused imports

#### files.py:
- Remove copy.copy, datetime.(date,timedelta), unused imports
- Convert exceptions to strings before adding strings to them

#### gui.py:
- Add default value for returnText, could possibly be undefined
- changed hidden -> disabled (not 100% sure this was needed)

#### operations.py:
- Remove scripts.logging import, unused imports

#### prepare_modes.py:
- Remove Scrips.confusablesCustom.normalize, base64, pathlib, unused imports

#### requirements.txt:
- Updated versions of packages to latest, untested though

#### utils.py:
- Added clear_terminal function to use instead of os.system everywhere in YTSpammerPurge.py, more logic behind it

#### YTSpammerPurge.py:
- remove ast and platform, unused imports
- add comments describing what S B & F do, one letter variables aren't always the nicest to read
- replace clear_command instances with utils.clear_terminal()
- remove redundant ast.literal_eval(str(json.load(file))) -> just json.load(file)
- remove redundant check for if userChoice == False: validInteger == False (validInteger set to False by default)
- remove redundant redeclaration of validConfigSetting, it will be set two lines below anyway
- fix bug with dict as default value, set it instead every time function is called (I think)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# Additional Info
I am not able to test these changes at the moment, but I do not think any of these would be breaking changes

## Checklist:

- [x] My code follows the style guidelines of this project and have read CONTRIBUTING.md
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings *(untested)*
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by micziz — APPROVED** [review]
OK? Don't really understand

**Comment by EthanHindmarsh** [comment]
RE: micziz
Mostly just removing unused imports

**Comment by micziz** [comment]
Ok

**Review by Firecul — APPROVED** [review]
Readme and config changes look good to me. 
I'm not qualified to judge the rest but atm I don't *see* anything breaking.

File: Scripts/utils.py [hunk 1]
```diff
@@ -213,3 +213,15 @@ def print_error_title_fetch():
   print(f"  > You won't be able to delete/hide any comments like usual, but you can {F.LIGHTMAGENTA_EX}exclude users before saving the log file{S.R}")
   print(f"  > Then, you can {F.LIGHTGREEN_EX}delete the comments later{S.R} using the {F.YELLOW}mode that removes comments using a pre-existing log file{S.R}")
   input("\n Press Enter to continue...")
+
+def clear_terminal() -> None:
```
Comment (compsup) [inline Scripts/utils.py:217]: Is there a reason you have made this function way more complicated than it needs to be? For example why use `sys.stdout.write("\033[2J\033[1;1H")` when you could just do `os.system("clear")`.

**Review by compsup — COMMENTED** [review]
Just one thing too look at.

File: Scripts/utils.py (hunk 1 above)
Comment (EthanHindmarsh) [reply Scripts/utils.py:217]: I'd think sys.stdout.write() (practically just print) would be at least a tiny bit faster than os.system()

Comment (EthanHindmarsh) [reply Scripts/utils.py:217]: But no significant difference other than os.system() runs a command and sys.stdout.write() does not

Comment (Firecul) [reply Scripts/utils.py:217]: It's faster according to here https://www.geeksforgeeks.org/clear-console-c-language/

Comment (EthanHindmarsh) [reply Scripts/utils.py:217]: As for "more complicated", there are only two `if`s in clear_terminal():

if it is not a tty (ex. the script is running as a GUI application or is being piped into another program or file), do not clear anything and return

if windows: os.system(cls)
else: write "\033[2J\033[1;1H"

As a bonus, `clear_command` is no longer a variable, and therefore no arbitrary os.system() call could be made by changing `clear_command`; clear_terminal() is easier to read 😁

**Review by compsup — APPROVED** [review]

**Review by KendallDoesCoding — APPROVED** [review]

File: Scripts/utils.py (hunk 1 above)
Comment (EthanHindmarsh) [reply Scripts/utils.py:217]: Oh also, in the future, a config option could be added to disable clearing the terminal (probably dev_mode or something) by either redefining clear_terminal() or adding some check to clear_terminal()

File: README.md [hunk 2]
```diff
@@ -1,36 +1,36 @@
 <h1 align="center">
-<img src="/assets/icon.png" alt="Icon" width="100" height="100" </img>
+<img src="/assets/icon.png" alt="Icon" style="width:100px;height:100px;"/>
```
Comment (KendallDoesCoding) [inline README.md:2]: Well it's find as it is, in my opinion, style isn't really needed.

File: README.md [hunk 3]
```diff
@@ -1,36 +1,36 @@
...
+    <img alt="Code Quality status" src="https://github.com/ThioJoe/YT-Spammer-Purge/actions/workflows/codeql-analysis.yml/badge.svg" />
   </a>
   <a href="https://github.com/ThioJoe/YT-Spammer-Purge/actions/workflows/docker-build.yml">
-    <img src="https://github.com/ThioJoe/YT-Spammer-Purge/actions/workflows/docker-build.yml/badge.svg" />
+    <img alt="Build status" src="https://github.com/ThioJoe/YT-Spammer-Purge/actions/workflows/docker-build.yml/badge.svg" />
   </a>
 </div>
 
-**What Is This?** - Allows you to filter and search for spammer comments on your channel and other's channel(s) in many different ways AND delete/report them all at once (see features below).
+**What Is This?** - Allows you to filter and search for spammer comments on your channel and other's channel(s) in many ways AND delete/report them all at once (see features below).
```
Comment (KendallDoesCoding) [inline README.md:26]: I mean, different kinda adds some effect, I would keep it as is, but there's no harm if you remove the word too, but just my opinion.

File: README.md [hunk 4]
```diff
@@ -57,15 +57,15 @@ YouTube Spammer Purge
 
 ## Purpose
 
-Recently, there has been a massive infestation of spam on YouTube where fake impersonator accounts leave spam/scam replies to hundreds of users on a creator's videos. For some god-forsaken reason, YouTube offers no way to delete all comments by a specific user at once, meaning you must delete them one by one BY HAND.
+Recently, there has been a massive infestation of spam on YouTube where fake impersonator accounts leave spam/scam replies to hundreds of users on a creator's videos. For some godforsaken reason, YouTube offers no way to delete all comments by a specific user at once, meaning you must delete them BY HAND.
```
Comment (KendallDoesCoding) [inline README.md:60]:
```suggestion
Recently, there has been a massive infestation of spam on YouTube where fake impersonator accounts leave spam/scam replies to hundreds of users on a creator's videos. For some godforsaken reason, YouTube offers no way to delete all comments by a specific user at once, meaning you must delete them individually BY HAND.
```

**Comment by ThioJoe** [comment]
I was going through this and there are a few things I want to revert which is kind of a pain, could you resubmit with _only_ these changes below included? I'd do it myself but I figured you'd want the PR credit.

- Update to requirements.txt
- Update to install.sh
- Removal of the various unused imports / redundant variables
- The e to str(e) change
- Comment about S,B,F import
- The clear terminal stuff

The rest I have reasons to keep as is but if you are wondering about any specifics I could explain

**Comment by EthanHindmarsh** [comment]
Yes, it would not be hard to partially revert some of my changes👍
I will have that done in a bit

**Comment by EthanHindmarsh** [comment]
> The rest I have reasons to keep as is but if you are wondering about any specifics I could explain

I've left this change:

```diff
-  jsonData = open(spamListDict['Meta']['VersionInfo']['Path'], 'r', encoding="utf-8")
-  versionInfoJson = str(json.load(jsonData)) # Parses json file into a string
-  versionInfo = ast.literal_eval(versionInfoJson) # Parses json string into a dictionary
+  versionInfo = json.load(open(spamListDict['Meta']['VersionInfo']['Path'], 'r', encoding="utf-8"))
```

It was not part of the list you said to include, but it removes the import for `ast` and should work the same way 
Current execution is to

1.  Open file
2. Load json from file into python data and convert to string
3. Load string as python data

Change makes it just:
1. Load json from file into python data


Other than that, please re-review and let me know! :+1: @ThioJoe

**Comment by ThioJoe** [comment]
Basically for that one, I recall encountering a bunch of weirdness when it came to loading json data and converting it between json and dictionary, so I don't really want to touch that.

**Comment by EthanHindmarsh** [comment]
RE: Thio
The weirdness would've still been there when ast was used then, since
json.load() is used in both?
If you can show me how using ast fixes an issue with json.load(), I'll
understand 😅
But as is, it's just an extra step that slows converts the data from
json.load() into a str and back into data
I've never seen json have an issue, especially since it's in the standard
library

**Comment by ThioJoe** [comment]
Your code outputs a string, where I need a dictionary:
image

image

**Comment by EthanHindmarsh** [comment]
RE: Thio
That is odd, I will revert the change and commit
If everything else is fine this PR should be ready for merge👍

**Comment by EthanHindmarsh** [comment]
@ThioJoe Reverted change to JSON loading, I might check out why it does not work and open another PR about it 👍

**Comment by Firecul** [comment]
> Checking out file changes, seems one of my neovim hooks cleared the extra whitespace in YTSpammerPurge.py as well, though that doesn't affect anything😅

Ideally, it shouldn't have been there anyway tbf

**Comment by EthanHindmarsh** [comment]
Merge upstream to ensure no conflicts ^
@ThioJoe Ready for review whenever 👍

**Pull request merged**